service/history/workflowTaskHandler.go

```diff
@@ -117,14 +117,15 @@ func newWorkflowTaskHandler(
 	config *configs.Config,
 	shard shard.Context,
 	searchAttributesMapper searchattribute.Mapper,
+	hasBufferedEvents bool,
 ) *workflowTaskHandlerImpl {
 
 	return &workflowTaskHandlerImpl{
 		identity:                identity,
 		workflowTaskCompletedID: workflowTaskCompletedID,
 
 		// internal state
-		hasBufferedEvents:               mutableState.HasBufferedEvents(),
+		hasBufferedEvents:               hasBufferedEvents,
```

Comment:
Any case hasBufferedEvents != mutableState.HasBufferedEvents() ?

Reply:
`hasUnhandledEvents` which is passed here from the function in file bellow, gets his value from the same `ms.HasBufferedEvents()`. I need this parameter to support workflow sync update.